README
Программа OutWiker предназначена для хранения заметок в виде дерева. Такие программы называют outliner'ами, персональными вики или древовидными редакторами. Я долго подбирал для себя подобную программу из имеющихся, но в итоге решил сделать свой outliner в том виде, как он мне представляется наиболее удобным для использования.

Основное отличие OutWiker от других подобных программ заключается в том, что OutWiker не только хранит дерево заметок в виде обычных директорий на диске, но и поощряет изменение такой базы внешними средствами. Также к программе может быть прикреплено любое количество файлов.

Кроме того, OutWiker может содержать страницы разного типа, в данный момент поддерживаются три типа страниц: обычный текст, HTML и страница поиска по дереву, но в будущем количество типов страниц будет увеличиваться.

Особая благодарность разработчикам сайтов http://www.famfamfam.com, https://p.yusukekamiyamane.com и https://www.fatcow.com/free-icons за бесплатные иконки, которые используются в OutWiker'е.

Программа OutWiker распространяется с исходными текстами под лицензией GPL 3 (http://www.gnu.org/licenses/gpl.html).

И привычная для всех программ отговорка: "Автор не несет ответственность за последствия использования программы".

Автор: Евгений Ильин (aka Jenyay) <jenyay.ilin@gmail.com>
Сайт автора: https://jenyay.net


debian/changelog
outwiker (1.9.0+790~wily) wily; urgency=medium

  * Small fixes

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 07 Apr 2016 22:34:35 +0300

outwiker (1.9.0+789~wily) wily; urgency=medium

  * Next version

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 10 Mar 2016 20:57:57 +0300

outwiker (1.9.0+788~wily) wily; urgency=medium

  * Added menu items and hot keys to moving to pages
  * Bug fizes

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 10 Mar 2016 19:30:17 +0300

outwiker (1.9.0+787~wily) wily; urgency=medium

  * Next version

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 23 Feb 2016 14:54:55 +0300

outwiker (1.9.0+786~wily) wily; urgency=medium

  * Bug fixes

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 22 Feb 2016 18:51:45 +0300

outwiker (1.9.0+785~wily) wily; urgency=medium

  * Next version

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 16 Feb 2016 20:49:17 +0300

outwiker (1.9.0+784~wily) wily; urgency=medium

  * Bug fixes

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 11 Feb 2016 09:29:41 +0300

outwiker (1.9.0+783~wily) wily; urgency=medium

  * Next version

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 08 Feb 2016 09:48:35 +0300

outwiker (1.9.0+782~wily) wily; urgency=medium

  * Added [!...!] wiki command.
  * Refactoring.

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Fri, 05 Feb 2016 10:04:59 +0300

outwiker (1.9.0+781~wily) wily; urgency=medium

  * Next version

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 01 Dec 2015 21:15:12 +0300

outwiker (1.9.0+780~wily) wily; urgency=medium

  * Fixed preferences dialog
  * Fixed crashes with spell checker and wiki colorizing

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 26 Nov 2015 20:43:24 +0300

outwiker (1.9.0+779~wily) wily; urgency=medium

  * Next version

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Fri, 30 Oct 2015 22:11:54 +0300

outwiker (1.9.0+778~wily) wily; urgency=medium

  * Acelerated icons drawing
  * Rafactoring PageDialog

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 29 Oct 2015 08:52:44 +0300

outwiker (1.9.0+777~wily) wily; urgency=medium

  * Acelerated icons drawing
  * Rafactoring PageDialog

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Fri, 18 Sep 2015 10:04:55 +0300

outwiker (1.9.0+776~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 17 Sep 2015 20:05:21 +0300

outwiker (1.9.0+775~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 24 Aug 2015 17:12:59 +0300

outwiker (1.9.0+774~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 24 Aug 2015 13:35:49 +0300

outwiker (1.9.0+773~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 27 Jul 2015 10:18:17 +0300

outwiker (1.9.0+772~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Sat, 25 Jul 2015 21:09:27 +0300

outwiker (1.9.0+771~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 23 Jun 2015 21:57:40 +0300

outwiker (1.9.0+770~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Sat, 20 Jun 2015 21:47:57 +0300

outwiker (1.9.0+769~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 21 May 2015 09:44:13 +0300

outwiker (1.9.0+768~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 19 May 2015 22:39:06 +0300

outwiker (1.9.0+767~vivid) vivid; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 26 Mar 2015 21:23:53 +0300

outwiker (1.9.0+766~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 23 Mar 2015 20:10:54 +0300

outwiker (1.9.0+765~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 09 Feb 2015 09:25:03 +0300

outwiker (1.9.0+764~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Sun, 08 Feb 2015 19:54:14 +0300

outwiker (1.9.0+763~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 22 Jan 2015 22:35:40 +0300

outwiker (1.9.0+762~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 19 Jan 2015 21:17:58 +0300

outwiker (1.9.0+761~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 09 Dec 2014 21:34:42 +0300

outwiker (1.9.0+760~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 09 Dec 2014 20:48:02 +0300

outwiker (1.9.0+759~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 17 Nov 2014 21:15:04 +0300

outwiker (1.9.0+758~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 17 Nov 2014 20:19:59 +0300

outwiker (1.9.0+757~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Sun, 16 Nov 2014 22:23:47 +0300

outwiker (1.9.0+756~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Sun, 16 Nov 2014 22:21:45 +0300

outwiker (1.9.0+755~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 28 Oct 2014 20:44:39 +0300

outwiker (1.9.0+754~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 28 Oct 2014 19:44:01 +0300

outwiker (1.9.0+753~utopic) utopic; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Mon, 20 Oct 2014 15:18:05 +0400

outwiker (1.8.1+752~trusty) trusty; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Fri, 17 Oct 2014 22:06:16 +0400

outwiker (1.8.0+751~trusty) trusty; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Tue, 14 Oct 2014 19:15:40 +0400

outwiker (1.8.0+750~trusty) trusty; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Thu, 09 Oct 2014 21:45:00 +0400

outwiker (1.8.0+745~trusty) trusty; urgency=medium

  * 

 -- Eugeniy Ilin <jenyay.ilin@gmail.com>  Sat, 04 Oct 2014 19:04:27 +0400

outwiker (1.8.0+744~trusty) trusty; urgency=medium

  * 


src/gui/about.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# generated by wxGlade 0.6.3 on Thu Nov 18 21:58:50 2010

import wx

# begin wxGlade: extracode
# end wxGlade



class AboutDialog(wx.Dialog):
	def __init__(self, currversion, *args, **kwds):
		# begin wxGlade: AboutDialog.__init__
		kwds["style"] = wx.DEFAULT_DIALOG_STYLE|wx.RESIZE_BORDER|wx.THICK_FRAME
		wx.Dialog.__init__(self, *args, **kwds)
		self.notebook = wx.Notebook(self, -1, style=0)
		self.donatePane = wx.Panel(self.notebook, -1)
		self.aboutPane = wx.Panel(self.notebook, -1)
		self.titleLabel = wx.StaticText(self, -1, _("OutWiker"))
		self.label_1 = wx.StaticText(self, -1, _("Version:"))
		self.versionLabel = wx.StaticText(self, -1, "")
		self.label_2 = wx.StaticText(self.aboutPane, -1, _("Author: Ilin Eugeniy (aka Jenyay)"))
		self.label_3 = wx.StaticText(self.aboutPane, -1, _("License: GPL 3"))
		self.label_4 = wx.StaticText(self.aboutPane, -1, _("My site:"))
		self.siteLink = wx.HyperlinkCtrl(self.aboutPane, -1, label=_("http://jenyay.net"), url=_("http://jenyay.net"))
		self.label_5 = wx.StaticText(self.aboutPane, -1, _("OutWiker's page:"))
		self.outwikerPage = wx.HyperlinkCtrl(self.aboutPane, -1, label=_("http://jenyay.net/Soft/Outwiker"), url=_("http://jenyay.net/Soft/Outwiker"))
		self.label_6 = wx.StaticText(self.aboutPane, -1, _("Twitter:"))
		self.twitterLink = wx.HyperlinkCtrl(self.aboutPane, -1, label=_("@jenyay"), url=_("http://twitter.com/#!/jenyay"))
		self.label_7 = wx.StaticText(self.aboutPane, -1, _("E-mail:"))
		self.mailLink = wx.HyperlinkCtrl(self.aboutPane, -1, label=_("jenyay.ilin@gmail.com"), url=_("mailto:jenyay.ilin@gmail.com"))
		self.label_8 = wx.StaticText(self.donatePane, -1, _("WebMoney (WMR):"))
		self.wmrTextCtrl = wx.TextCtrl(self.donatePane, -1, _("R356551209135"), style=wx.TE_READONLY)
		self.label_10 = wx.StaticText(self.donatePane, -1, _("WebMoney (WMZ)"))
		self.wmzTextCtrl = wx.TextCtrl(self.donatePane, -1, _("Z427584680984"))
		self.label_9 = wx.StaticText(self.donatePane, -1, _("Yandex.Money"))
		self.yandexTextCtrl = wx.TextCtrl(self.donatePane, -1, _("41001580829031"), style=wx.TE_READONLY)
		self.okButton = wx.Button(self, wx.ID_OK, "")

		self.__set_properties()
		self.__do_layout()
		# end wxGlade

		self.currversion = currversion
		self.versionLabel.SetLabel (str (self.currversion))


	def __set_properties(self):
		# begin wxGlade: AboutDialog.__set_properties
		self.SetTitle(_("About"))
		self.titleLabel.SetFont(wx.Font(15, wx.DEFAULT, wx.NORMAL, wx.NORMAL, 0, ""))
		self.okButton.SetFocus()
		self.okButton.SetDefault()
		# end wxGlade

	def __do_layout(self):
		# begin wxGlade: AboutDialog.__do_layout
		main_sizer = wx.FlexGridSizer(4, 1, 0, 0)
		grid_sizer_7 = wx.FlexGridSizer(3, 2, 0, 0)
		grid_sizer_2 = wx.FlexGridSizer(6, 1, 0, 0)
		grid_sizer_6 = wx.FlexGridSizer(1, 2, 0, 0)
		grid_sizer_5 = wx.FlexGridSizer(1, 2, 0, 0)
		grid_sizer_4 = wx.FlexGridSizer(1, 2, 0, 0)
		grid_sizer_3 = wx.FlexGridSizer(1, 2, 0, 0)
		grid_sizer_1 = wx.FlexGridSizer(1, 2, 0, 0)
		main_sizer.Add(self.titleLabel, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|wx.ALIGN_CENTER_VERTICAL, 2)
		grid_sizer_1.Add(self.label_1, 0, wx.ALL|wx.ALIGN_RIGHT, 2)
		grid_sizer_1.Add(self.versionLabel, 0, wx.ALL, 2)
		grid_sizer_1.AddGrowableRow(0)
		grid_sizer_1.AddGrowableCol(0)
		grid_sizer_1.AddGrowableCol(1)
		main_sizer.Add(grid_sizer_1, 1, wx.EXPAND, 0)
		grid_sizer_2.Add(self.label_2, 0, wx.ALL, 2)
		grid_sizer_2.Add(self.label_3, 0, wx.ALL, 2)
		grid_sizer_3.Add(self.label_4, 0, wx.ALL, 2)
		grid_sizer_3.Add(self.siteLink, 1, wx.ALL|wx.EXPAND, 2)
		grid_sizer_3.AddGrowableCol(1)
		grid_sizer_2.Add(grid_sizer_3, 1, wx.EXPAND, 0)
		grid_sizer_4.Add(self.label_5, 0, wx.ALL, 2)
		grid_sizer_4.Add(self.outwikerPage, 1, wx.ALL|wx.EXPAND, 2)
		grid_sizer_4.AddGrowableCol(1)
		grid_sizer_2.Add(grid_sizer_4, 1, wx.EXPAND, 0)
		grid_sizer_5.Add(self.label_6, 0, wx.ALL, 2)
		grid_sizer_5.Add(self.twitterLink, 1, wx.ALL|wx.EXPAND, 2)
		grid_sizer_5.AddGrowableCol(1)
		grid_sizer_2.Add(grid_sizer_5, 1, wx.EXPAND, 0)
		grid_sizer_6.Add(self.label_7, 0, wx.ALL, 2)
		grid_sizer_6.Add(self.mailLink, 1, wx.ALL|wx.EXPAND, 2)
		grid_sizer_6.AddGrowableCol(1)
		grid_sizer_2.Add(grid_sizer_6, 1, wx.EXPAND, 0)
		self.aboutPane.SetSizer(grid_sizer_2)
		grid_sizer_2.AddGrowableCol(0)
		grid_sizer_7.Add(self.label_8, 0, wx.ALL|wx.ALIGN_CENTER_VERTICAL, 2)
		grid_sizer_7.Add(self.wmrTextCtrl, 0, wx.ALL|wx.EXPAND, 2)
		grid_sizer_7.Add(self.label_10, 0, wx.ALL|wx.ALIGN_CENTER_VERTICAL, 2)
		grid_sizer_7.Add(self.wmzTextCtrl, 0, wx.ALL|wx.EXPAND, 2)
		grid_sizer_7.Add(self.label_9, 0, wx.ALL|wx.ALIGN_CENTER_VERTICAL, 2)
		grid_sizer_7.Add(self.yandexTextCtrl, 0, wx.ALL|wx.EXPAND, 2)
		self.donatePane.SetSizer(grid_sizer_7)
		grid_sizer_7.AddGrowableCol(0)
		grid_sizer_7.AddGrowableCol(1)
		self.notebook.AddPage(self.aboutPane, _("About"))
		self.notebook.AddPage(self.donatePane, _("Donate"))
		main_sizer.Add(self.notebook, 1, wx.ALL|wx.EXPAND, 2)
		main_sizer.Add(self.okButton, 0, wx.ALL|wx.ALIGN_RIGHT|wx.ALIGN_CENTER_VERTICAL, 4)
		self.SetSizer(main_sizer)
		main_sizer.Fit(self)
		main_sizer.AddGrowableRow(2)
		main_sizer.AddGrowableCol(0)
		self.Layout()
		self.Centre()
		# end wxGlade

# end of class AboutDialog


